### DISCUSSION TITLE
Visualizing and Comparing Embedding Vectors as Heatmaps | Tanel Poder Consulting

### DISCUSSION THREAD

**Discussion by giscus** [tanelpoder/blog-comments#44] [Announcements]
# Visualizing and Comparing Embedding Vectors as Heatmaps | Tanel Poder Consulting

If you already know what embeddings are, you can jump right to:
Vector Heatmap Visualization Background I am currently testing various RDBMS vector search options and their usability &amp; performance. Next week I will publish an early version of my CatBench toolset that will eventually evolve into a full end-to-end test (and fun!) suite for high performance AI/ML pipelines, with things like streaming image recognition using Vision Transformers on GPUs, followed by vector searches in target databases and more. - Linux, Oracle, SQL performance tuning and troubleshooting - consulting & training.

https://tanelpoder.com/posts/visualizing-embedding-vectors-as-heatmaps/

**Comment by aifartist** [comment]
I did something like this long ago. What I choose to analyse was fruits vs vegetables. I wanted to see if "tomato" was closer in the embedding space to fruits or vegetables. I also looked at which embedding indexes had the largest absolute values and whether fruit shared common indexes among themselves and independently for vegetables. And, yes, there were indexes in common which stood out as illustrated by a heat map.
